### PR TITLE
Testing fslexyacc caching + update devguide

### PR DESCRIPTION
**Pull request by vzarytovskii**
Just testing the waters here, not entirely sure it's a good idea yet, maybe we should stick with some custom task to check the cache validity.

Parser and lexer outputs are cached in the repository and are regenerated automatically every time fslexyacc input files are changed (checked by fslexyacc itself).

However, they will not be regenerated if any dependent compiler libraries are changed.
If you are making changes to the latter, and do not see your changes in the parser/lexer, you will need to manually regenerate the outputs.
This can be done by removing `src\.autogenerated\fslexyacc` directory, and running build again.

Previously, it was enough to just clean your git folder (`git clean -xdf` of similar) or remove obj directory (or freshly clone the repo).

On my machine, this change saves roughly 20-25 seconds when building the compiler, if no changes were made to the lexer/parser.

Any thoughts about this? An alternative would be a custom task which hashes fslexyacc itself, its inputs and dependant compiler files and skips invoking fslexyacc alltogether if nothing changes, this will allow to save even more (by not invoking fslexyacc at all).

### CODE REVIEW

File: DEVGUIDE.md [hunk 1]
```diff
@@ -134,6 +159,11 @@ To use your custom build of `Fsc`, add the `DotnetFscCompilerPath` property to y
 </PropertyGroup>
 ```
 
+## Updating parser and lexer
+
+Parser and lexer outputs are cached in the repository and are regenerated automatically every time fslexyacc input files are changed (fsl and fsy files), however, they will not be regenerated if any dependent compiler libraries are changed.
+If you are making changes to the latter, and do not see your changes in the parser/lexer, you will need to manually regenerate the outputs. This can be done by removing `src\.autogenerated\fslexyacc` directory, and running build again.
```
Comment (nojaf) [inline DEVGUIDE.md:165]: Is there a command (something `dotnet msbuild` maybe) we can run instead to regenerate this?

Comment (vzarytovskii) [reply DEVGUIDE.md:165]: Not oob, maybe FsLex/FsYacc tasks can be modified to check some flag.

File: buildtools/fsyacc/fsyacc.fs [hunk 2]
```diff
@@ -56,12 +56,15 @@ let main() =
...
-  printfn "        building tables"
-  printfn "        %d states" compiledSpec.states.Length; 
-  printfn "        %d nonterminals" compiledSpec.gotoTable.[0].Length; 
-  printfn "        %d terminals" compiledSpec.actionTable.[0].Length; 
-  printfn "        %d productions" compiledSpec.prods.Length; 
-  printfn "        #rows in action table: %d" compiledSpec.actionTable.Length; 
+
+  // Printing is nopt useful for day-to-day compiler dev and was just taking time and space in output.
+  // Enable if needed.
+  //printfn "        building tables"
```
Comment (nojaf) [inline buildtools/fsyacc/fsyacc.fs:62]: I'm not particularly fond that you are not introducing a change in https://github.com/fsprojects/FsLexYacc first.
By doing this you are deviating from the source and nobody will ever dare to sync them again in the future.

Also, make this configurable, and disable it by default.

Comment (vzarytovskii) [reply buildtools/fsyacc/fsyacc.fs:62]: This is to assess if it even makes sense to make this change or not

**Comment by vzarytovskii** [comment]
This need a full revamp, closing.

**Pull request closed without merging**